### PR TITLE
doc: DX update for GSG

### PR DESCRIPTION
**Pull request by dbkinder**
- Update the Getting Started material with a DX-inspired rewrite and
  simplification.
- Remove duplicate and out-of-date "Building from Source"
  document, deferring to the new GSG.
- Add a development overview document.
- Move other GSGs to the advanced guides section.
- Update links in other documents to aim at the new GSG.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>
Signed-off-by: Amy Reyes <amy.reyes@intel.com>

### CODE REVIEW

File: doc/getting-started/overview_dev.rst [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+For more information about the kernel, see :ref:`kernel-parameters`.
+
+.. _overview_dev_install:
+
+|icon_target| Step 6: Install and Run ACRN
+******************************************
+
+The last step is to make final changes to the target system configuration and
+boot ACRN.
```
Comment (fitchbe) [inline doc/getting-started/overview_dev.rst:281]: Consider "and then boot ACRN" here instead of "and boot ACRN".

File: doc/try.rst [hunk 2]
```diff
@@ -3,21 +3,19 @@
...
 
    reference/hardware
+   getting-started/overview_dev
    getting-started/getting-started
-   getting-started/building-from-source
-   getting-started/roscube/roscube-gsg
-   tutorials/using_hybrid_mode_on_nuc
-   tutorials/using_partition_mode_on_nuc
+
+After getting familiar with ACRN development, Check out these
```
Comment (fitchbe) [inline doc/try.rst:19]: Typo (should be a lowercase _c_ in _check_).

File: doc/try.rst [hunk 3]
```diff
@@ -3,21 +3,19 @@
...
+:ref:`develop_acrn` for information about more advanced scenarios and enabling
```
Comment (fitchbe) [inline doc/try.rst:20]: "more advanced scenarios" --> "more-advanced scenarios"

**Review by fitchbe — APPROVED** [review]

**Comment by dbkinder** [comment]
OK to verify

**Review by gvancuts — APPROVED** [review]
LGTM

**Pull request merged**